src/services/school/hooks/index.js

```diff
@@ -199,8 +199,18 @@ const restrictToUserSchool = async (context) => {
 
 const populateInQuery = (context) => (context.params.query || {}).$populate;
 
+const isNotAuthenticated = async (context) => {
+	if (typeof (context.params.provider) === 'undefined') {
+		return false;
+	} else {
+		return !((context.params.headers || {}).authorization || context.params.account && context.params.account.userId);
```

Comment:
Codacy found an issue: [Unexpected mix of '||' and '&&'.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5734878)